include/RAJA/internal/Span.hpp

```diff
@@ -88,18 +88,18 @@ struct Span {
                 "ValueType must model RandomAccessIterator");
 
   iterator begin() { return _data; }
-  iterator end() { return std::advance(_data, _length); }
+  iterator end() { return _data + _length; }
   const_iterator begin() const { return _data; }
-  const_iterator end() const { return std::advance(_data, _length); }
+  const_iterator end() const { return _data + _length; }
```

Comment:
it's even more obvious -- `advance()` has no return type and mutates the first parameter